homeassistant/components/hive/water_heater.py

```diff
@@ -62,19 +82,31 @@ def temperature_unit(self):
     @property
     def current_operation(self):
         """Return current operation."""
-        return HIVE_TO_HASS_STATE[self.session.hotwater.get_mode(self.node_id)]
+        return HIVE_TO_HASS_STATE[self.device["status"]["current_operation"]]
 
     @property
     def operation_list(self):
         """List of available operation modes."""
         return SUPPORT_WATER_HEATER
 
     @refresh_system
-    def set_operation_mode(self, operation_mode):
+    async def async_turn_on(self, **kwargs):
+        """Turn on hotwater."""
+        await self.hive.hotwater.set_mode(self.device, "MANUAL")
+
+    @refresh_system
+    async def async_turn_off(self, **kwargs):
+        """Turn on hotwater."""
+        await self.hive.hotwater.set_mode(self.device, "OFF")
+
+    @refresh_system
+    async def async_set_operation_mode(self, operation_mode):
         """Set operation mode."""
         new_mode = HASS_TO_HIVE_STATE[operation_mode]
-        self.session.hotwater.set_mode(self.node_id, new_mode)
+        await self.hive.hotwater.set_mode(self.device, new_mode)
 
-    def update(self):
+    async def async_update(self):
         """Update all Node data from Hive."""
-        self.session.core.update_data(self.node_id)
+        await self.hive.session.updateData(self.device)
+        self.device = await self.hive.hotwater.get_hotwater(self.device)
+        self.attributes.update(self.device.get("attributes", {}))
```

Comment:
The attributes aren't used for this entity.

Reply:
Fixed in latest commit